tests/test_discovery.py

```diff
@@ -514,6 +515,22 @@ def test_patch(self):
 
     self.assertEqual(request.method, 'PATCH')
 
+  def test_batch_request_from_discovery(self):
+    self.http = HttpMock(datafile('zoo.json'), {'status': '200'})
+    # zoo defines a batchPath
+    zoo = build('zoo', 'v1', http=self.http)
+    expected = "https://www.googleapis.com/batchZoo"
+    actual = zoo.new_batch_http_request()
```

Comment:
I'm wary of identifiers that are only adjectives. This is a BatchHttpRequest, right? How about calling it "batch_request"?